include/grpc/event_engine/posix.h

```diff
@@ -0,0 +1,158 @@
+// Copyright 2022 gRPC Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#ifndef GRPC_EVENT_ENGINE_POSIX_H
+#define GRPC_EVENT_ENGINE_POSIX_H
+
+#include <grpc/support/port_platform.h>
+
+#include <functional>
+#include <vector>
+
+#include "absl/functional/any_invocable.h"
+#include "absl/status/status.h"
+
+#include <grpc/event_engine/endpoint_config.h>
+#include <grpc/event_engine/event_engine.h>
+#include <grpc/event_engine/memory_allocator.h>
+#include <grpc/event_engine/port.h>
+#include <grpc/event_engine/slice_buffer.h>
+
+namespace grpc_event_engine {
+namespace experimental {
+
+#ifdef GRPC_EVENT_ENGINE_POSIX
+
+/// This defines an EventEngine interface that all posix specific event engines
+/// must implement.
```

Comment:
I would say `may implement`. It's not an explicit requirement for all engines, but is required to support certain functionality. All of our posix-friendly engines will likely implement it, but this is a public API.

Reply:
fixed